q2_mlab/tests/test_preprocessing.py

```diff
@@ -51,17 +52,49 @@ def test_preprocess_output(self):
                                   discrete=True,
                                   with_replacement=False,
                                   n_jobs=1)
-        self.assertEqual(len(results), 8)
+        self.assertEqual(len(results), 9)
         self.assertTrue(str(results[0].type) == 'FeatureTable[Frequency]')
-        self.assertTrue(str(results[1].type) == 'SampleData[Target]')
-        self.assertTrue(str(results[2].type) == 'DistanceMatrix')
+        self.assertTrue(str(results[1].type) == 'FeatureTable[Frequency]')
+        self.assertTrue(str(results[2].type) == 'SampleData[Target]')
         self.assertTrue(str(results[3].type) == 'DistanceMatrix')
         self.assertTrue(str(results[4].type) == 'DistanceMatrix')
         self.assertTrue(str(results[5].type) == 'DistanceMatrix')
+        self.assertTrue(str(results[6].type) == 'DistanceMatrix')
 
         phylo_dm_string = "DistanceMatrix % Properties('phylogenetic')"
-        self.assertTrue(str(results[6].type) == phylo_dm_string)
         self.assertTrue(str(results[7].type) == phylo_dm_string)
+        self.assertTrue(str(results[8].type) == phylo_dm_string)
+
+        self.assertEqual(
+            results[0].view(biom.Table).shape[1],
+            results[1].view(biom.Table).shape[1]
+        )
+
+    def test_preprocess_output_no_phylo(self):
+
+        results = self.preprocess(table=self.mp_table,
+                                  metadata=self.mp_sample_metadata,
+                                  sampling_depth=1000,
+                                  min_frequency=10,
+                                  target_variable="body-site",
+                                  discrete=True,
+                                  with_replacement=False,
+                                  n_jobs=1)
+        self.assertEqual(len(results), 9)
+        self.assertTrue(str(results[0].type) == 'FeatureTable[Frequency]')
+        self.assertTrue(str(results[1].type) == 'FeatureTable[Frequency]')
+        self.assertTrue(str(results[2].type) == 'SampleData[Target]')
+        self.assertTrue(str(results[3].type) == 'DistanceMatrix')
+        self.assertTrue(str(results[4].type) == 'DistanceMatrix')
+        self.assertTrue(str(results[5].type) == 'DistanceMatrix')
+        self.assertTrue(str(results[6].type) == 'DistanceMatrix')
+        self.assertTrue(str(results[7].type) == 'DistanceMatrix')
+        self.assertTrue(str(results[8].type) == 'DistanceMatrix')
```

Comment:
Can you load these two objects and check they have a shape 0, 0?

Reply:
Added in a check. When I create the "empty" distance matrices, a (1,1) matrix is the smallest I can go:
```
In [9]: dm = DistanceMatrix(data=[])                                                                                                                                                                                           

In [10]: dm.data                                                                                                                                                                                                               
Out[10]: array([[0.]])
```
The saved weighted_unifrac_distance_matrix.qza and unweighted_unifrac_distance_matrix.qza will all be 28kb, which is noticeably smaller than the others. Not able to rename the output files just for this condition AFAIK.